### PR TITLE
Misc minor fixes

### PR DESCRIPTION
**Pull request by aalemayhu**
@dborkmann I have gone through the bpf directory and tried to fix some of the __uX usage, but most of the changes ended up focusing on silencing the check output. @tgraf took care of the port related ones in c5c24504bf8e (bpf: Use __be16 consistently to indicate ports in network byteorder, 2017-08-29).

The output from sparse still seems mostly related to the macros, not sure how to fix them without breaking compilation. Do you see anything wrong or missed? Please let me know, thanks!

### CODE REVIEW

File: bpf/bpf_lxc.c [hunk 1]
```diff
@@ -258,8 +258,7 @@ static inline int ipv6_l3_from_lxc(struct __sk_buff *skb,
 		/* After L4 write in port mapping: revalidate for direct packet access */
 		data = (void *) (long) skb->data;
 		data_end = (void *) (long) skb->data_end;
-		ip6 = data + ETH_HLEN;
-		if (data + sizeof(*ip6) + ETH_HLEN > data_end)
+		if (data + sizeof(*(data + ETH_HLEN)) + ETH_HLEN > data_end)
```
Comment (aanm) [inline bpf/bpf_lxc.c]: I think it would help to have a comment before the line so developers know what is supposed to be the `data + ETH_HLEN`

Comment (tgraf) [inline bpf/bpf_lxc.c]: I think we can remove this length check altogether, it looks like code that was accessing ip6 before was removed without removing this check.

**Review by tgraf — CHANGES_REQUESTED** [review]

**Review by tgraf — APPROVED** [review]

**Comment by tgraf** [comment]
@scanf Is this just an ordering issue with github? I see the `bpf.sha` update commit in the middle of the series

**Comment by aalemayhu** [comment]
I think the web UI is sorting on date, which is incorrect. Because of rebasing.
One easy way to know what the last commit is independent of the UI order is to look for the CI indicator (green check mark, yellow circle, etc), which shows you the last commit on the CI. 

Squashed the commit to avoid confusion.

**Pull request merged**